docs/src/main/sphinx/functions/json.rst

```diff
@@ -2,6 +2,1152 @@
+
+``json_input`` is a character string or a binary string. It should contain
+a single JSON item. For a binary string, you can specify encoding.
+
+``json_path`` is a string literal, containing the path mode specification, and
+the path expression, following the syntax rules described in
+:ref:`path_syntax_and_semantics`.
+
+.. code-block:: text
+
+    'strict $.keyvalue()?(@.name == $cust_id)'
+    'lax $[5 to last]'
+
+In the ``PASSING`` clause you can pass arbitrary expressions to be used by the
+path expression.
+
+.. code-block:: text
+
+    PASSING orders.custkey AS CUST_ID
+
+The passed parameters can be referenced in the path expression by named
+variables, prefixed with ``$``.
+
+.. code-block:: text
+
+    'strict $.keyvalue()?(@.value == $CUST_ID)'
+
+Additionally to SQL values, you can pass JSON values, specifying the format and
+optional encoding:
+
+.. code-block:: text
+
+    PASSING orders.json_desc FORMAT JSON AS o_desc,
+            orders.binary_record FORMAT JSON ENCODING UTF16 AS o_rec
+
+Note that the JSON path language is case-sensitive, while the unquoted SQL
+identifiers are upper-cased. Therefore, it is recommended to use quoted
+identifiers in the ``PASING`` clause:
+
+.. code-block:: text
+
+    'lax $.$KeyName' PASSING nation.name AS KeyName --> ERROR; no passed value found
+    'lax $.$KeyName' PASSING nation.name AS "KeyName" --> correct
+
+The ``ARRAY WRAPPER`` clause lets you modify the output by wrapping the results
+in a JSON array. ``WITHOUT ARRAY WRAPPER`` is the default option. ``WITH
+CONDITIONAL ARRAY WRAPPER`` wraps every result which is not a singleton JSON
+array or JSON object. ``WITH UNCONDITIONAL ARRAY WRAPPER`` wraps every result.
+
+The ``QUOTES`` clause lets you modify the result for a scalar strings by
+removing the double quotes being part of the JSON string representation.
+
+Examples
+^^^^^^^^
+
+Let ``customers`` be a table containing two columns: ``id:bigint``,
+``description:varchar``.
+
+========== ==================================================
+id         description
+========== ==================================================
+101        '{"comment" : "nice", "kids" : [10, 13, 16]}'
+102        '{"comment" : "problematic", "kids" : [8, 11]}'
+103        '{"comment" : "knows best", "kids" : [2]}'
+========== ==================================================
+
+The following query gets the ``kids`` array for each customer:
+
+.. code-block:: text
+
+    SELECT id, json_query(
+                          description,
+                          'lax $.kids'
+                         ) kids
+    FROM customers
+
+========== ================
+id         kids
+========== ================
+101        '[10,13,16]'
+102        '[8,11]'
+103        '[2]'
+========== ================
+
+The following query gets the collection of kids for each customer. Note that
+the ``json_query`` function can only output a single JSON item. If we don't use
```

Comment:
If you don't use ... probably need to scrub the whole doc to avoid "we" in usage like here .. just think about talking to a reader informally ..